diff --git a/drivers/staging/rtl8712/rtl871x_xmit.c b/drivers/staging/rtl8712/rtl871x_xmit.c
index 8bbdee70..e2266e90 100644
--- a/drivers/staging/rtl8712/rtl871x_xmit.c
+++ b/drivers/staging/rtl8712/rtl871x_xmit.c
@@ -208,13 +208,13 @@ sint r8712_update_attrib(struct _adapter *padapter, _pkt *pkt,
 	pattrib->ether_type = ntohs(etherhdr.h_proto);
 
 {
-	u8 bool;
+	u8 Bool;
 	/*If driver xmit ARP packet, driver can set ps mode to initial
 	 * setting. It stands for getting DHCP or fix IP.*/
 	if (pattrib->ether_type == 0x0806) {
 		if (padapter->pwrctrlpriv.pwr_mode !=
 		    padapter->registrypriv.power_mgnt) {
-			_cancel_timer(&(pmlmepriv->dhcp_timer), &bool);
+			_cancel_timer(&(pmlmepriv->dhcp_timer), &Bool);
 			r8712_set_ps_mode(padapter, padapter->registrypriv.
 				power_mgnt, padapter->registrypriv.smart_ps);
 		}
diff --git a/include/linux/stddef.h b/include/linux/stddef.h
index 6a40c76b..7b3e42dc 100644
--- a/include/linux/stddef.h
+++ b/include/linux/stddef.h
@@ -12,11 +12,6 @@
 
 #ifdef __KERNEL__
 
-enum {
-	false	= 0,
-	true	= 1
-};
-
 #undef offsetof
 #ifdef __compiler_offsetof
 #define offsetof(TYPE,MEMBER) __compiler_offsetof(TYPE,MEMBER)
diff --git a/include/linux/types.h b/include/linux/types.h
index 57a97234..767ea4a4 100644
--- a/include/linux/types.h
+++ b/include/linux/types.h
@@ -15,6 +15,7 @@
 #endif
 
 #include <linux/posix_types.h>
+#include <stdbool.h>
 
 #ifdef __KERNEL__
 
@@ -34,8 +35,6 @@ typedef __kernel_timer_t	timer_t;
 typedef __kernel_clockid_t	clockid_t;
 typedef __kernel_mqd_t		mqd_t;
 
-typedef _Bool			bool;
-
 typedef __kernel_uid32_t	uid_t;
 typedef __kernel_gid32_t	gid_t;
 typedef __kernel_uid16_t        uid16_t;
diff --git a/kernel/panic.c b/kernel/panic.c
index b2659360..c9c67232 100644
--- a/kernel/panic.c
+++ b/kernel/panic.c
@@ -160,8 +160,8 @@ EXPORT_SYMBOL(panic);
 
 struct tnt {
 	u8	bit;
-	char	true;
-	char	false;
+	char	True;
+	char	False;
 };
 
 static const struct tnt tnts[] = {
@@ -211,7 +211,7 @@ const char *print_tainted(void)
 		for (i = 0; i < ARRAY_SIZE(tnts); i++) {
 			const struct tnt *t = &tnts[i];
 			*s++ = test_bit(t->bit, &tainted_mask) ?
-					t->true : t->false;
+					t->True : t->False;
 		}
 		*s = 0;
 	} else
